### PR TITLE
Mesa 7.0 branch

### PR DESCRIPTION
**Pull request by uli42**
this obsoletes the mesa70 PR #529 

Mesa has no tag for the 7.0.4 release, so we use the latest commit in
the mesa_7_0_branch (which is 77a8e26bf824b45c1244f5699d7cbfca0fe0a48a).

Unfortunately we are going _down_ version wise:

Building nx-libs with Mesa 7.0 results in glxinfo reporting:
  OpenGL version string: 1.2 (2.1 Mesa 7.0)
But building with Mesa 7.0.4 results in
  OpenGL version string: 1.2 (2.0 Mesa 7.0.4)

Reason: 7.0.4 disables one OpenGL extension essential for OpenGL 2.1
called GL_ARB_shading_language_120 (because it is not feature
complete).

### CODE REVIEW

File: nx-X11/programs/Xserver/GL/glx/indirect_dispatch.c [hunk 1]
```diff
@@ -25,7 +25,7 @@
  * SOFTWARE.
  */
 
-#include <X11/Xmd.h>
+#include <nx-X11/Xmd.h>
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/GL/glx/indirect_dispatch.c]: @uli42: shouldn't this commit (48a080b384fc8168eddae8542990ed9e181de96e) be pulled over to the small_fixes PR #532 ?

**Comment by sunweaver** [comment]
@uli42: my sense is to skip the mesa 6.5 PR and directly jump to this one... (or even to a newer one)?

**Comment by uli42** [comment]
I am at 7.7 at the moment. 8.0 is difficult due to missing foundations. I tend to take latest 7.x and integrate that. Next is upgrading the server to around xorg 1.8.0. Then we should be able to go on with mesa.

Vom Smartphone gesendet.

----- Ursprüngliche Nachricht -----
Von: "Mike Gabriel" <notifications@github.com>
Gesendet: ‎21.‎10.‎2017 10:46
An: "ArcticaProject/nx-libs" <nx-libs@noreply.github.com>
Cc: "Ulrich Sibiller" <uli42@gmx.de>; "Mention" <mention@noreply.github.com>
Betreff: Re: [ArcticaProject/nx-libs] Mesa 7.0 branch (#531)

@uli42: my sense is to skip the mesa 6.5 PR and directly jump to this one... (or even to a newer one)?

**Comment by sunweaver** [comment]
Hi Uli,

On  Sa 21 Okt 2017 11:25:13 CEST, Ulrich Sibiller wrote:

> I am at 7.7 at the moment. 8.0 is difficult due to missing  
> foundations. I tend to take latest 7.x and integrate that. Next is  
> upgrading the server to around xorg 1.8.0. Then we should be able to  
> go on with mesa.

Ok, let's go for Mesa 7.7 then. For the merging process, I guess it  
would be good if you give me access to the PR branch. So that I can  
commit fixes if needed directly into the PR itself.

Get back to me, when any of the branches is ready for a second glimpse.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by uli42** [comment]
We don't have that 'indirect_dispatch.c' file in 3.6.x And if we had it would come from Mesa
and belong to the Mesa patchset.

**Comment by sunweaver** [comment]
will be superceded by #534

**Pull request closed without merging**